truffleruby/src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -702,4 +705,68 @@ protected ForeignToRubyNode createForeignToRubyNode() {
         }
 
     }
+
+    @CoreMethod(names = "polyglot_in_read", isModuleFunction = true, required = 1, lowerFixnum = 1)
+    public abstract static class PolyglotInReadNode extends CoreMethodArrayArgumentsNode {
+
+        @TruffleBoundary
+        @Specialization
+        public DynamicObject polyglotInRead(int length) {
+            byte[] buffer = new byte[length];
+
+            final int read;
+
+            try {
+                read = getContext().getEnv().in().read(buffer);
+            } catch (IOException e) {
+                throw new JavaException(e);
+            }
+
+            if (read == -1) {
+                return nil();
+            }
+
+            if (read < buffer.length) {
+                buffer = Arrays.copyOf(buffer, read);
+            }
+
+            return createString(buffer, USASCIIEncoding.INSTANCE);
```

Comment:
It should be ASCIIEncoding then (without US).
But don't trust Ruby docs:
```
$ ruby -e 'p gets.encoding'  
dfsdf
#<Encoding:UTF-8>
```
I would think it's using the external encoding.
Anyway this behavior already exists somewhere if we reuse the normal calls.